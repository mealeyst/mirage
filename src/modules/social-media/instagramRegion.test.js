import React from 'react'
import 'jest-styled-components'

import { InstagramRegion, H1, P } from 'SRC'

const { mountWithTheme } = global

describe('(Styled Component) InstagramRegion', () => {
  const defaultProps = {
    title: 'Example Title',
    content: 'This is the example content'
  }
  const createInstagramRegion = (props = defaultProps) => {
    return mountWithTheme(<InstagramRegion {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createInstagramRegion())
    .toMatchSnapshot()
  })

  test('rendering the title', () => {
    const title = 'Hello World'
    expect(
      createInstagramRegion({...defaultProps, title: title})
      .find(H1)
      .text()
    ).toEqual(title)
  })

  test('rendering the content', () => {
    const content = 'An updated content area'
    expect(
      createInstagramRegion({...defaultProps, content: content})
      .find(P)
      .text()
    ).toEqual(content)
  })
})
