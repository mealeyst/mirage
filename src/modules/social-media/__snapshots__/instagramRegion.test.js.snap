// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) InstagramRegion matching the snapshot 1`] = `
.c2 {
  margin: 0;
  -webkit-letter-spacing: .1rem;
  -moz-letter-spacing: .1rem;
  -ms-letter-spacing: .1rem;
  letter-spacing: .1rem;
  text-transform: uppercase;
  color: #00003C;
  font-family: "din-cond","din-condensed-web",Arial,sans-serif;
  font-size: 4.2rem;
  font-weight: 500;
  line-height: 1.0476190476190477;
}

.c4 {
  text-align: left;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  color: #00003C;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: 1.6rem;
  line-height: normal;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 > a {
  width: 50%;
  max-width: 17rem;
  padding: 1rem;
  box-sizing: border-box;
}

.c5 > a img {
  width: 100%;
  max-width: 15rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 > * {
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .c1,
.c0 .c3 {
  margin-bottom: 2rem;
  text-align: center;
}

<Styled(BaseInstagramRegion)
  content="This is the example content"
  title="Example Title"
>
  <BaseInstagramRegion
    className="c0"
    content="This is the example content"
    title="Example Title"
  >
    <section
      className="c0"
    >
      <styled.h1
        fontSize="4.2rem"
      >
        <h1
          className="c1 c2"
          fontSize="4.2rem"
        >
          Example Title
        </h1>
      </styled.h1>
      <styled.p
        fontSize="1.6rem"
      >
        <p
          className="c3 c4"
          fontSize="1.6rem"
        >
          This is the example content
        </p>
      </styled.p>
      <Styled(BaseInstagram)
        limit={4}
      >
        <BaseInstagram
          className="c5"
          limit={4}
        >
          <section
            className="c5"
          />
        </BaseInstagram>
      </Styled(BaseInstagram)>
    </section>
  </BaseInstagramRegion>
</Styled(BaseInstagramRegion)>
`;
